app/controllers/posts_controller.rb

```diff
@@ -1,15 +1,51 @@
 class PostsController < ApplicationController
 
+  before_action :find_post, only: [:show, :edit, :update]
+
   def index
     @posts = Post.order(created_at: :desc)
   end
 
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    if @post.save
+      redirect_to :root
+    else
+      render :new
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if @post.update(post_params)
+      redirect_to @post
+    else
+      render :edit
+    end
+  end
+
   def show
-    @post = Post.find(params[:id])
     @user = @post.user
```

Comment:
Why set the instance variable here when you can always do `@post.user` in the view?